pkg/controllers/resources/ingresses/syncer.go

```diff
@@ -137,7 +137,12 @@ func (s *ingressSyncer) applyLimitByClass(ctx *synccontext.SyncContext, virtual
 			s.EventRecorder().Eventf(virtual, "Warning", "SyncWarning", "did not sync ingress %q to host because the ingress class %q couldn't be reached in the host: %s", virtual.GetName(), *virtual.Spec.IngressClassName, err)
 			return true
 		}
-		if !ctx.Config.Sync.FromHost.IngressClasses.Selector.Matches(pIngressClass) {
+		matches, err := ctx.Config.Sync.FromHost.IngressClasses.Selector.Matches(pIngressClass)
+		if err != nil {
+			s.EventRecorder().Eventf(virtual, "Warning", "SyncWarning", "did not sync ingress %q to host because the ingress class %q in the host could not be checked against the selector under 'sync.fromHost.ingressClasses.selector': %s", virtual.GetName(), pIngressClass.GetName(), err)
```

Comment:
I was almost making a function to make this less repetitive.
Please tell me if I should make a function or a "method" to `syncertypes.GenericTranslator` (a bit agressive).